### PR TITLE
announce map titles

### PR DESCRIPTION
**Pull request by fabiangreffrath**
https://www.doomworld.com/forum/topic/125256-nugget-doom-310-updated-may-16th-24/?page=22&tab=comments#comment-2801367

### CODE REVIEW

**Comment by MrAlaux** [comment]
Well, that's handy. Works just the way I envisioned it. Thanks!

**Comment by fabiangreffrath** [comment]
Alternatively, we could show it as a centered message, just like the secret revealed message but with the automap's title color. But this would look stupid if you open the automap immediately after entering the level and have the title displayed twice.

**Comment by MrAlaux** [comment]
> Alternatively, we could show it as a centered message, just like the secret revealed message but with the automap's title color. But this would look stupid if you open the automap immediately after entering the level and have the title displayed twice.

Couldn't the automap title simply be disabled while the centered one is being shown?

File: src/hu_stuff.c [hunk 1]
```diff
@@ -168,6 +168,10 @@ boolean           show_messages;
 boolean           show_toggle_messages;
 boolean           show_pickup_messages;
 
+boolean           hud_map_announce;
```
Comment (rfomin) [inline src/hu_stuff.c:171]: `hud_map_announce` should be `static`

**Review by rfomin — COMMENTED** [review]
What if we annonce map title with regular message, like in DSDA-Doom?
doom00

**Comment by MrAlaux** [comment]
> What if we annonce map title with regular message, like in DSDA-Doom?

That can be easily overridden by other messages (e.g. picking something up), which I think is not desirable.

**Comment by fabiangreffrath** [comment]
> > What if we annonce map title with regular message, like in DSDA-Doom?
> 
> That can be easily overridden by other messages (e.g. picking something up), which I think is not desirable.

Yes, right.

**Review by rfomin — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
> Couldn't the automap title simply be disabled while the centered one is being shown?

On second thought, I'd find it misleading if the map title was ever displayed at a different position then the one chosen for the automap.

**Pull request merged**